lib/database/classes.ts

```diff
@@ -63,37 +79,141 @@ const updateClass = async (
+        rrstring: classesWithUserInformation[i].rrstring,
+        startTime: classesWithUserInformation[i].startTime,
+        endTime: classesWithUserInformation[i].endTime,
+        language: classesWithUserInformation[i].language,
+      };
+      individualClasses.push(currClass);
+      const curr = {
+        userId: classesWithUserInformation[i].userId,
+        firstName: classesWithUserInformation[i].firstName,
+        lastName: classesWithUserInformation[i].lastName,
+      };
+      const teachersArr = [curr];
+      mapOfClasses.set(classesWithUserInformation[i].eventInformationId, teachersArr);
+    } else {
+      const curr = {
+        userId: classesWithUserInformation[i].userId,
+        firstName: classesWithUserInformation[i].firstName,
+        lastName: classesWithUserInformation[i].lastName,
+      };
+      const teachersArr = mapOfClasses.get(classesWithUserInformation[i].eventInformationId);
+      teachersArr.push(curr);
 
-  return oneClass;
+      mapOfClasses.set(classesWithUserInformation[i].eventInformationId, teachersArr);
+    }
+  }
+  for (const singleClass of individualClasses) {
+    const currClass: Class = {
+      name: singleClass.name,
+      eventInformationId: singleClass.eventInformationId,
+      minLevel: singleClass.minLevel,
+      maxLevel: singleClass.maxLevel,
+      rrstring: singleClass.rrstring,
+      startTime: singleClass.startTime,
+      endTime: singleClass.endTime,
+      language: singleClass.language,
+      teachers: mapOfClasses.get(singleClass.eventInformationId),
+    };
+    return currClass;
+  }
+  return null;
 };
 const getAllClasses = async (): Promise<Class[]> => {
   const query = {
-    text: "SELECT e.name, c.event_information_id, c.min_level, c.max_level, c.rrstring, c.start_time, c.end_time, c.language FROM event_information e, classes c WHERE e.id = c.event_information_id AND e.type = 'Class'",
+    text:
+      "SELECT e.name, cl.event_information_id, cl.min_level, cl.max_level, cl.rrstring, cl.start_time, cl.end_time, cl.language, u.id as user_id, u.first_name, u.last_name " +
+      "FROM (((event_information e INNER JOIN classes cl ON e.id = cl.event_information_id) " +
+      "INNER JOIN commitments ON commitments.event_information_id = e.id) " +
+      " INNER JOIN users u ON commitments.user_id = u.id) WHERE role = 'Teacher'",
   };
-
   const res = await client.query(query);
-
-  let classesArray: classArrayType;
+  //console.log(res);
+  let classesWithUserInformation: ClassWithUserInformation[];
   try {
-    classesArray = await decode(ClassArraySchema, res.rows);
+    classesWithUserInformation = await decode(ClassWithUserInformationArraySchema, res.rows);
   } catch {
     throw Error("Fields returned incorrectly in database");
   }
+  const classesArray: Class[] = [];
+  const mapOfClasses = new Map();
+  const individualClasses = [];
+  for (let i = 0; i < classesWithUserInformation.length; i++) {
```

Comment:
instead of using for loops, can we use a more functional `.forEach()`. Take a look at the JS documentation